File: todo-mobile/android-kotlin/app/src/main/kotlin/co/thehindu/todonative/Todo.kt
package co.thehindu.todonative

import org.json.JSONObject

data class Todo(
    val id: Int,
    val title: String,
    val completed: Boolean,
) {
    companion object {
        fun fromJson(json: JSONObject): Todo = Todo(
            id = json.getInt("id"),
            title = json.getString("title"),
            completed = json.optBoolean("completed", false),
        )
    }
}

File: todo-mobile/mobile/lib/models/todo.dart
class Todo {
  final int id;
  final String title;
  final bool completed;

  const Todo({required this.id, required this.title, required this.completed});

  factory Todo.fromJson(Map<String, dynamic> json) => Todo(
        id: json['id'] as int,
        title: json['title'] as String,
        completed: json['completed'] as bool? ?? false,
      );

  Todo copyWith({String? title, bool? completed}) => Todo(
        id: id,
        title: title ?? this.title,
        completed: completed ?? this.completed,
      );
}
